week2/homework/src/index.js

```diff
@@ -1,3 +1,82 @@
-'use strict';
+'use strict:';
 
-// TODO: Write the homework code in this file
+const fs = require('fs');
+const { promisify } = require('util');
+const dataToNewLine = data => `\n${data}`;
+
+const writeFile = promisify(fs.writeFile);
+const readfile = promisify(fs.readFile);
+
+const filePath = './ToDoList.json';
+
+const getAllToDos = () => {
+  return readfile(filePath, 'utf8')
+    .then(data => JSON.parse(data))
+    .catch(err => console.log(err));
+};
+
+const updateTodo = async (id, newTitle) => {
+  const { todos } = await getAllToDos();
+  const updatedTodos = todos.map(todo => {
+    if (todo.id === id) {
+      return {
+        id,
+        title: newTitle,
+      };
+    }
+    return todo;
+  });
```

Comment:
Wouldn't this be simpler and easier to read?
```javascript
let toUpdate = todos.find(todo => todo.id === id);
toUpdate.title = newTitle;
```

Reply:
I will try this method. 👍